ecs-cli/modules/clients/aws/cloudformation/template.go

```diff
@@ -470,50 +461,16 @@ var template = `
           "Ref": "EcsInstanceProfile"
         },
         "KeyName": {
-          "Ref": "KeyName"
-        },
-        "SecurityGroups": {
           "Fn::If": [
-            "CreateSecurityGroup",
-            [ {
-              "Ref": "EcsSecurityGroup"
-            } ],
+            "CreateEC2LCWithKeyPair",
             {
-              "Ref": "SecurityGroupIds"
+              "Ref": "KeyName"
+            },
+            {
+              "Ref": "AWS::NoValue"
             }
           ]
         },
-        "UserData": {
-          "Fn::Base64": {
-            "Fn::Join": [
-              "",
-              [
-                "#!/bin/bash\n",
-                "echo ECS_CLUSTER=",
-                {
-                  "Ref": "EcsCluster"
-                },
-                " >> /etc/ecs/ecs.config\n"
-              ]
-            ]
-          }
-        }
-      }
-    },
-    "EcsInstanceLcWithoutKeyPair": {
-      "Condition": "CreateEC2LCWithoutKeyPair",
-      "Type": "AWS::AutoScaling::LaunchConfiguration",
-      "Properties": {
-        "ImageId": { "Ref" : "EcsAmiId" },
-        "InstanceType": {
-          "Ref": "EcsInstanceType"
-        },
-        "AssociatePublicIpAddress": {
-          "Ref": "AssociatePublicIpAddress"
-        },
-        "IamInstanceProfile": {
-          "Ref": "EcsInstanceProfile"
-        },
```

Comment:
May be I'm reading the diff wrong, but why are all these lines being removed?

Reply:
Previously, we had a condition to either create `EcsInstanceLc` if a keypair exists or a `EcsInstanceLcWithoutKeyPair` if it didn't. The latter resource was actually never created bc we were requiring keypair as a parameter.  This is now refactored to be similar to the cloudformation template used by the console team, which, rather than having a condition that decides which entire LaunchConfiguration resource gets created, just conditionally sets the KeyName field on one Launch Configuration template.

Reply:
Cool 👍

Reply:
Oh, also, I don't know if you already tested this, but make sure that this change doesn't break backwards compatibility with old cloudformation templates and cause the down command to not work. Other than that PR looks good and I'll approve.